virttest/utils_test/libvirt.py

```diff
@@ -3543,28 +3543,58 @@ def update_vm_disk_source(
         return False
 
 
-def exec_virsh_edit(source, edit_cmd, connect_uri="qemu:///system"):
+def exec_virsh_edit(
+    source,
+    edit_cmd,
+    connect_uri="qemu:///system",
+    managedsave_edit=False,
+    readonly=False,
+    virsh_opt="",
+):
     """
     Execute edit command.
 
     :param source : virsh edit's option.
     :param edit_cmd: Edit command list to execute.
+    :param connect_uri: Connection URI.
+    :param managedsave_edit: Use 'managedsave-edit' instead of 'edit'.
+    :param readonly: add -r for the virsh cmd
+    :param virsh_opt: the option for virsh cmd
     :return: True if edit is successful, False if edit is failure.
     """
     LOG.info("Trying to edit xml with cmd %s", edit_cmd)
     session = aexpect.ShellSession("sudo -s")
     try:
-        session.sendline("virsh -c %s edit %s" % (connect_uri, source))
+        command = "managedsave-edit" if managedsave_edit else "edit"
+        if readonly:
+            session.sendline(
+                "virsh -r -c %s %s %s %s" % (connect_uri, command, source, virsh_opt)
+            )
+        else:
+            session.sendline(
+                "virsh -c %s %s %s %s" % (connect_uri, command, source, virsh_opt)
+            )
         for cmd in edit_cmd:
             session.sendline(cmd)
         session.send("\x1b")
         session.send("ZZ")
-        remote.handle_prompts(session, None, None, r"[\#\$]\s*$", debug=True)
+        # Capture output until the next shell prompt
```

Comment:
Why did you substitute the call to `handle_prompts`, looks like a risky change that might introduce regression. Please elaborate in your commit message why you changed this and what you think the involved risk is and if necessary, what you did to mitigate that risk.

Reply:
The handle_prompts automates interaction with a shell to get it to a state where it's ready for the next command. With `session.read_until_last_line_matches([prompt_pattern], timeout=10)`, it directly waits for the shell prompt (# or $). It also captures all the output from the virsh edit command into the output variable. It will then inspect the output, which is more explicit and robust.

Reply:
As Sebastian highlight here, it needs to do some testing validating whether  remote.handle_prompts can be replaced equally with session.read_until_last_line_matches.